doc/decisions/key_string_return_value.md

```diff
@@ -0,0 +1,38 @@
+# keyString() return value
+
+## Problem
+
+When using keyString() on empty / binary values the return values are the literal strings (null) / (binary). This seems very awkward and unintuitive from a user's perspective.
+
+## Constraints
+
+Because the return value of the function is const char\*, there are not so many possibilities for a change to this behaviour, as every possible return value - except for a NULL pointer - could just be the value of the string of the Key. Possible changes to this behaviour would be:
```

Comment:
Actually we could also change the return value, so this is not really a constraint.

Reply:
I think we can say "we don't want to change the return value", because that would make the function very inconvenient to use.

Reply:
"return value" is an error from the original text. Probably changing the API is meant?